frontend/src/components/Topics/Topic/Messages/MessagesTable.tsx

```diff
@@ -28,8 +57,79 @@ const MessagesTable: React.FC<MessagesTableProps> = ({
   const nextCursor = useMessageFiltersStore((state) => state.nextCursor);
   const isLive = useIsLiveMode();
 
+  const [selectedFormat, setSelectedFormat] = useState<DownloadFormat>('json');
+
+  const formatOptions: SelectOption<DownloadFormat>[] = [
+    { label: 'JSON', value: 'json' },
+    { label: 'CSV', value: 'csv' }
+  ];
+
+  const handleFormatSelect = (format: DownloadFormat) => {
+    setSelectedFormat(format);
+  };
+
+  const handleDownload = () => {
+
+  const savedMessagesJson: MessageData[] = messages.map(message => ({
+    Value: message.content,
+    Offset: message.offset,
+    Key: message.key,
+    Partition: message.partition,
+    Headers: message.headers,
+    Timestamp: message.timestamp,
+  }));
+
+  const convertToCSV = (messages: MessageData[]) => {
+    const headers = ['Value', 'Offset', 'Key', 'Partition', 'Headers', 'Timestamp'] as const;
+
+    const rows = messages.map(msg =>
+      headers.map(header => {
+        const value = msg[header];
+        if (header === 'Headers') {
+          return JSON.stringify(value || {});
+        }
+        return String(value ?? '');
+      }).join(',')
+    );
+
+    return [headers.join(','), ...rows].join('\n');
+  };
+
+    const baseFileName = 'topic-messages'+padCurrentDateTimeString();
+    const jsonSaver = useDataSaver(`${baseFileName}.json`, JSON.stringify(savedMessagesJson, null, '\t'));
+    const csvSaver = useDataSaver(`${baseFileName}.csv`, convertToCSV(savedMessagesJson));
+
+    if (selectedFormat === 'json') {
+      jsonSaver.saveFile();
+    } else {
+      csvSaver.saveFile();
+    }
+  };
+
   return (
     <div style={{ position: 'relative' }}>
+      <div style={{ display: 'flex', gap: '8px', marginLeft: '1rem', marginBottom: '1rem' }}>
+        <Select<DownloadFormat>
+          id="download-format"
+          name="download-format"
+          onChange={handleFormatSelect}
+          options={formatOptions}
+          value={selectedFormat}
+	  minWidth="70px"
+	  selectSize="M"
+          placeholder="Select format to download"
+          disabled={isFetching || messages.length === 0}
+        />
+        <Button
+          disabled={isFetching || messages.length === 0}
+          buttonType="secondary"
+          buttonSize="M"
+          onClick={handleDownload}
+        >
+          Download All Messages
```

Comment:
The text `Download All Messages` may be a bit misleading because this will only download the messages of the current page. In the future we may want to expose an actual "Download all the messages" that will paginate and download the messages

Reply:
will do.

Reply:
done